src/test/ui/libstd-case-typo/char.rs

```diff
@@ -0,0 +1,17 @@
+// checks case typos with libstd::char structs
+fn main(){}
+
+fn test_du16(_x: DecodeUTF16<()>){}
+//~^ ERROR: cannot find type `DecodeUTF16` in this scope
+
+fn test_edflt(_x: Escapedefault){}
```

Comment:
I'd combine all of these tests into a single file - we're not gaining anything by having them separate - if we change the approach, it's very likely that if one fails then they all would and so having different tests just means we have more tests to run (likely to be marginally slower).